cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java

```diff
@@ -550,6 +550,36 @@ public void testStopWhilePending() throws Exception {
     }
   }
 
+  @Test
+  public void testStopProgramRunWhilePending() throws Exception {
+    try {
+      deploy(SleepingWorkflowApp.class, 200, Constants.Gateway.API_VERSION_3_TOKEN, TEST_NAMESPACE1);
+      Id.Program workflowId =
+        Id.Program.from(TEST_NAMESPACE1, SLEEP_WORKFLOW_APP_ID, ProgramType.WORKFLOW, SLEEP_WORKFLOW_NAME);
+
+      int numKilledRuns = getProgramRuns(workflowId, ProgramRunStatus.KILLED).size();
+
+      // this tells the provisioner to wait for 60s before trying to create the cluster for the run
+      Map<String, String> args = new HashMap<>();
+      args.put(SystemArguments.PROFILE_PROPERTIES_PREFIX + MockProvisioner.WAIT_CREATE_MS, Integer.toString(120000));
+      startProgram(workflowId, args);
+
+      // should be safe to wait for starting since the provisioner is configure to sleep while creating a cluster
+      waitState(workflowId, co.cask.cdap.proto.ProgramStatus.STARTING.name());
+      List<RunRecord> runRecords = getProgramRuns(workflowId, ProgramRunStatus.PENDING);
+      Assert.assertEquals(1, runRecords.size());
+      String runId = runRecords.iterator().next().getPid();
+
+      stopProgram(workflowId, runId, 200);
+      waitState(workflowId, STOPPED);
+
+      verifyProgramRuns(workflowId, ProgramRunStatus.KILLED, numKilledRuns);
+    } finally {
+      Assert.assertEquals(200, doDelete(getVersionedAPIPath("apps/" + SLEEP_WORKFLOW_APP_ID, Constants.Gateway
+        .API_VERSION_3_TOKEN, TEST_NAMESPACE1)).getStatusLine().getStatusCode());
```

Comment:
This indentation looks a bit awkward, maybe can extract the result of doDelete() to a variable to better indent the line